hs_app_timeseries/models.py

```diff
@@ -228,96 +498,348 @@ def get_xml(self, pretty_print=True):
+                                             element_fields)
 
         return etree.tostring(RDF_ROOT, pretty_print=True)
 
     def delete_all_elements(self):
         super(TimeSeriesMetaData, self).delete_all_elements()
-        if self.site:
-            self.site.delete()
-        if self.variable:
-            self.variable.delete()
-        if self.method:
-            self.method.delete()
-        if self.processing_level:
-            self.processing_level.delete()
-        if self.time_series_result:
-            self.time_series_result.delete()
-
-
-import receivers
+        # delete resource specific metadata
+        self.sites.delete()
+        self.variables.delete()
+        self.methods.delete()
+        self.processing_levels.delete()
+        self.time_series_results.delete()
+        # delete CV lookup django tables
+        self.cv_variable_types.all().delete()
+        self.cv_variable_names.all().delete()
+        self.cv_speciations.all().delete()
+        self.cv_elevation_datums.all().delete()
+        self.cv_site_types.all().delete()
+        self.cv_method_types.all().delete()
+        self.cv_units_types.all().delete()
+        self.cv_statuses.all().delete()
+        self.cv_mediums.all().delete()
+        self.cv_aggregation_statistics.all().delete()
+
+    def update_sqlite_file(self):
+        if not self.is_dirty:
+            return
+
+        sqlite_file_to_update = self.resource.files.first()
+        if sqlite_file_to_update is not None:
+            istorage = IrodsStorage()
+            log = logging.getLogger()
+
+            # retrieve the sqlite file from iRODS and save it to temp directory
+            temp_dir = tempfile.mkdtemp()
+            sqlite_file_name = os.path.basename(sqlite_file_to_update.resource_file.name)
+            temp_sqlite_file_destination = os.path.join(temp_dir, sqlite_file_name)
+            istorage.getFile(sqlite_file_to_update.resource_file.name, temp_sqlite_file_destination)
+            try:
+                con = sqlite3.connect(temp_sqlite_file_destination)
+                with con:
+                    # get the records in python dictionary format
+                    con.row_factory = sqlite3.Row
+                    cur = con.cursor()
+                    self._update_variables_table(con, cur)
+                    self._update_methods_table(con, cur)
+                    self._update_processinglevels_table(con, cur)
+                    self._update_sites_related_tables(con, cur)
+                    self._update_results_related_tables(con, cur)
+                    self._update_CV_tables(con, cur)
+
+                    # push the updated sqlite file to iRODS
+                    sqlite_file_original = self.resource.files.first()
+                    to_file_name = sqlite_file_original.resource_file.name
+                    from_file_name = temp_sqlite_file_destination
+                    istorage.saveFile(from_file_name, to_file_name, True)
```

Comment:
@pkdash Writing files back to iRODS needs to take iRODS federation into account too. You can call https://github.com/hydroshare/hydroshare/blob/develop/hs_core/hydroshare/resource.py#L187 update_resource_file() method, but unfortunately this method is not updated to take into account iRODS federation yet. Can you modify that method to take into account iRODS federation by referring to https://github.com/hydroshare/hydroshare/blob/develop/hs_core/hydroshare/resource.py#L931 method and then call update_resource_file() method here rather than doing it here in this resource type?